lib/cli-utils.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-console */
+
+const JSON_SPACES = 4;
+
+/***
+ * Log an error to the console and exit the process.
+ * @param {boolean} json - whether we should log json or text
+ * @param {string} msg - error message
+ */
+function errAndQuit (json, msg) {
+  if (json) {
+    console.log(JSON.stringify({error: `${msg}`}), null, JSON_SPACES);
+  } else {
+    console.error(`${msg}`.red);
+  }
+  process.exit(1);
+}
+
+/**
+ * Conditionally log something to the console
+ * @param {boolean} json - whether we are in json mode (and should therefore not log)
+ * @param {string} msg - string to log
+ */
+function log (json, msg) {
+  !json && console.log(msg);
+}
+
+/**
+ * Start a CLI spinner
+ * @param {string} msg - what to label the spinner with
+ */
+function startSpinner (msg) {
```

Comment:
Would it make sense to use a package like [ora](https://www.npmjs.com/package/ora) here?

Reply:
ora looks great! i didn't know about it. i'll see if i can switch to use it.